src/Dfc.CourseDirectory.Web/Controllers/CopyCourse/CopyCourseRunController.cs

```diff
@@ -262,243 +219,355 @@ private async Task<SelectVenueModel> GetVenuesByUkprn(int ukprn)
+                StartDateType = courseRun.FlexibleStartDate
+                    ? StartDateType.FlexibleStartDate
+                    : StartDateType.SpecifiedStartDate,
+                Day = courseRun.StartDate?.Day.ToString("00"),
+                Month = courseRun.StartDate?.Month.ToString("00"),
+                Year = courseRun.StartDate?.Year.ToString("0000"),
+                StudyMode = courseRun.StudyMode,
+                Url = courseRun.CourseURL,
+                Cost = courseRun.Cost?.ToString("F"),
+                CostDescription = courseRun.CostDescription,
+                AttendanceMode = courseRun.AttendancePattern,
+                QualificationType = course.Value.QualificationType,
+                NotionalNVQLevelv2 = course.Value.NotionalNVQLevelv2,
+                PublishMode = PublishMode.Summary,
+                RefererAbsolutePath = Request.GetTypedHeaders().Referer?.AbsolutePath
+            };
+
+            if (savedModel != null)
+            {
+                vm.CourseName = savedModel.CourseName;
+                vm.CourseProviderReference = savedModel.CourseProviderReference;
+                vm.DeliveryMode = savedModel.DeliveryMode;
+                vm.StartDateType = savedModel.StartDateType;
+                vm.Day = savedModel.Day;
+                vm.Month = savedModel.Month;
+                vm.Year = savedModel.Year;
+                vm.VenueId = savedModel.VenueId;
+                vm.ChooseRegion.National = savedModel.National;
+                vm.Url = savedModel.Url;
+                vm.Cost = savedModel.Cost;
+                vm.CostDescription = savedModel.CostDescription;
+                vm.DurationLength = savedModel.DurationLength;
+                vm.DurationUnit = savedModel.DurationUnit;
+                vm.AttendanceMode = savedModel.AttendanceMode;
+                vm.StudyMode = savedModel.StudyMode;
+                vm.RefererAbsolutePath = savedModel.RefererAbsolutePath;
+
+                if (savedModel.DeliveryMode == DeliveryMode.ClassroomBased)
+                {
+                    foreach (var venue in vm.Venues)
                     {
-                        foreach (var subRegionItemModel in selectRegionRegionItem.SubRegion)
-                        {
-                            if (courseRun.Regions.Contains(subRegionItemModel.Id))
-                            {
-                                subRegionItemModel.Checked = true;
-                            }
-                        }
+                        venue.Selected = venue.Value == savedModel.VenueId.ToString();
                     }
+                }
 
-                    if (vm.ChooseRegion.Regions.RegionItems != null && vm.ChooseRegion.Regions.RegionItems.Any())
+                if (savedModel.DeliveryMode == DeliveryMode.WorkBased && !savedModel.National)
+                {
+                    foreach (var subRegion in regions?.RegionItems.SelectMany(r => r.SubRegion))
```

Comment:
As above - could throw NRE if `regions` is `null`

Reply:
`Regions` should never be `null`, so the use of the null conditional operator here is misleading.